docs/modules/ROOT/pages/gsn-bouncers.adoc

```diff
@@ -1,151 +1,146 @@
-
-  // Defined on base class GSNBouncerBase
-  // uint256 constant private RELAYED_CALL_REJECTED = 11;
-  // function _rejectRelayedCall(uint256 errorCode) internal pure returns (uint256, bytes memory) {
-  //    return (RELAYED_CALL_REJECTED + errorCode, "");
-  // }
+// Defined on base class GSNBouncerBase
+// uint256 constant private RELAYED_CALL_REJECTED = 11;
+// function _rejectRelayedCall(uint256 errorCode) internal pure returns (uint256, bytes memory) {
+//     return (RELAYED_CALL_REJECTED + errorCode, "");
+// }
 ----
 
+=== How to use GSNBouncerSignature
 
-=== How to use it
+You will need to create an off-chain service (e.g. backend server, microservice, or lambda function) that your dapp calls when requesting a GSN transaction, which signs (or doesn't sign) each transaction request based on your business logic.
 
-Create your contract using the following:
+Your GSN recipient contract needs to inherit from `GSNRecipient` and `GSNBouncerSignature`, as well as setting the trusted signer in the constructor of `GSNBouncerSignature` as per the following sample code below:
 
 [source,solidity]
 ----
-  mycontract is GSNRecipient, GSNBouncerSignature {
-    constructor(trusted_address) // that's it
-  }
+contract MyContract is GSNRecipient, GSNBouncerSignature {
+    constructor(address trustedSigner) public GSNBouncerSignature(trustedSigner) {
+    }
+}  
 ----
 
 == GSNBouncerERC20Fee
 
-This bouncer is a bit more complex (but don't worry, we've already written it for you!). Unlike `GSNBouncerSignature`, this Bouncer doesn't require any off-chain services.
-Instead of manually approving each transaction, you will give tokens to your users. These tokens are then used to pay for GSN calls to your recipient contract.
-Any user that has enough tokens is automatically approved and the recipient contract will cover his transaction costs!
+The https://docs.openzeppelin.com/contracts/2.x/api/gsn#GSNBouncerERC20Fee[`GSNBouncerERC20Fee`] is a bit more complex (but don't worry, it has already been written for you!). Unlike `GSNBouncerSignature`, the `GSNBouncerERC20Fee` doesn't require any off-chain services.
+Instead of off-chain approving each transaction, you will give contract specific ERC20 tokens to your users. These tokens are then used as payment for GSN transactions to your recipient contract.
+Any user that has enough tokens is automatically approved and the recipient contract will cover their transaction costs!
 
-This bouncer charges users for the ether cost your recipient will incur. Each recipient contract has their own unique token, with a baked-in exchange rate of 1:1 to ether, since they act as an ether replacement when using the GSN.
+This bouncer charges users for the ether cost your recipient contract will incur. Each recipient contract has their own unique token, with a baked-in exchange rate of 1:1 to ether, since they act as an ether replacement when using the GSN.
```

Comment:
I was confused by this paragraph. It repeats information from the previous line, and adds detail about the exchange rate without sufficiently explaining it.

I think it's worth trying to refactor most of the content in the `GSNBouncerERC20Fee` section into more distinct "introduction", "how it works", and "how to use it" subsections (not necessarily with subheaders, but at least as a way to structure the information). What do you think?

Similarly to `GSNBouncerSignature`, instead of showing the code snippets I would prefer if we explained in prose.

Reply:
Changed to prose.

Reply:
I have stripped out the code and written some prose.